src/jit-format/jit-format.cs

```diff
@@ -638,66 +639,67 @@ public static bool RunClangTidy(List<string> filenames, string compileCommands,
-                    if (verbose)
+                    formatOk = DoClangTidyInnerLoop(fix, ignoreErrors, checks, compileCommands, filename, verbose);
+                }
+            }
+            else
+            {
+                Parallel.ForEach(filenames, (filename) =>
                     {
-                        Console.WriteLine("Running: ");
-                        Console.WriteLine("\tclang-tidy " + tidyFix + " -checks=-*," + checks + fixErrors + " -header-filter=src/jit/.* -p " + compileCommands + " " + filename);
-                    }
+                        formatOk = DoClangTidyInnerLoop(fix, ignoreErrors, checks, compileCommands, filename, verbose);
+                    });
+            }
 
-                    Process process = new Process();
-                
-                    // Configure the process using the StartInfo properties.
-                    process.StartInfo.FileName = "clang-tidy";
-                    process.StartInfo.Arguments = tidyFix + "-checks=-*," + checks + fixErrors + " -header-filter=src/jit/.* -p " +
-                        compileCommands + " " + filename;
-                    process.StartInfo.UseShellExecute = false;
-                    process.StartInfo.RedirectStandardOutput = true;
-                    process.StartInfo.RedirectStandardError = true;
-                    process.StartInfo.CreateNoWindow = true;
-                    process.Start();
-
-                    string stdoutx = process.StandardOutput.ReadToEnd();         
-                    string stderrx = process.StandardError.ReadToEnd();             
-                    process.WaitForExit();
-
-
-                    if ((stdoutx.Contains("warning:") || (!ignoreErrors && stdoutx.Contains("error:"))))
-                    {
-                        if (verbose)
-                        {
-                            Console.WriteLine("clang-tidy: there are formatting errors in {0}", filename);
-                        }
-                        
-                        if (!fix)
-                        {
-                            formatOk = false;
-                        }
-                    }
+            return formatOk;
+        }
 
-                    if (!ignoreErrors && stderrx.Contains("error:"))
-                    {
-                        Console.Error.WriteLine("Error in clang-tidy: {0}", stderrx);
-                        
-                        if (!fix)
-                        {
-                            formatOk = false;
-                        }
-                    }
-                        
+        public static bool DoClangTidyInnerLoop(bool fix, bool ignoreErrors, string checks, string compileCommands, string filename, bool verbose)
+        {
+            string tidyFix = fix ? "-fix" : "";
+            string fixErrors = ignoreErrors && fix ? "-fix-errors" : "";
+
+            bool formatOk = true;
+
+            if (filename.EndsWith(".cpp"))
```

Comment:
Random note not related to this change: It's a bit unfortunate to hardcode this.  You don't need to change it since it's probably good enough for now, but it would be nice if this were derived from the compile_commands.json since that naturally filters out files that aren't directly compiled like headers.